SwiftGit2/References.swift

```diff
@@ -119,6 +120,27 @@ public struct Branch: ReferenceType {
 		}
 		commit = PointerTo<Commit>(oid)
 	}
+	
+	public func allCommits(in repo: Repository) -> [Commit] {
+		var commits: [Commit] = []
+		var walker: OpaquePointer? = nil
+		var unsafeCommit: OpaquePointer? = nil
+		var oid = self.oid.oid
+		git_revwalk_new(&walker, repo.pointer)
+		git_revwalk_sorting(walker, GIT_SORT_TOPOLOGICAL.rawValue)
+		git_revwalk_push(walker, &oid)
+		while git_revwalk_next(&oid, walker) == GIT_OK.rawValue {
+			guard
+				git_commit_lookup(&unsafeCommit, repo.pointer, &oid) == GIT_OK.rawValue,
+				let commit = unsafeCommit else {
+					return commits
```

Comment:
If `git_commit_lookup` fails to produce the requested commit, should the function return the commits, or should it return an error?